status/projects.py

```diff
@@ -597,34 +634,56 @@ class ProjectDataHandler(ProjectsBaseDataHandler):
 
     def get(self, project):
         self.set_header("Content-type", "application/json")
-        self.write(json.dumps(self.project_info(project)))
+        view_with_sources = self.get_argument("view_with_sources", False)
+        self.write(
+            json.dumps(self.project_info(project, view_with_sources=view_with_sources))
+        )
 
-    def project_info(self, project):
-        view = self.application.projects_db.view("project/summary")["open", project]
+    def project_info(self, project, view_with_sources=False):
+        if view_with_sources:
+            view_adress = "project/summary_with_sources"
+        else:
+            view_adress = "project/summary"
+        view = self.application.projects_db.view(view_adress)["open", project]
         if not view.rows:
-            view = self.application.projects_db.view("project/summary")[
-                "closed", project
-            ]
+            view = self.application.projects_db.view(view_adress)["closed", project]
         if not len(view.rows) == 1:
             return {}
 
         summary_row = view.rows[0]
-        summary_row = self.project_summary_data(summary_row)
+        summary_row, field_sources = self.project_summary_data(summary_row)
 
         date_view = self.application.projects_db.view(
             "project/summary_dates", descending=True, group_level=1
         )
         date_result = date_view[[project + "ZZZZ"] : [project]]
 
-        summary_row.value["_doc_id"] = summary_row.id
-
         if date_result.rows:
             for date_row in date_result.rows:
                 for date_type, date in date_row.value.items():
                     summary_row.value[date_type] = date
+                    field_sources[date_type] = "StatusDB view project/summary_dates"
+
+        summary_row.value["_doc_id"] = summary_row.id
+        field_sources["_doc_id"] = "StatusDB, inserted by Genomics Status (backend)"
         summary_row.value["sourcedb_url"] = (
             "https://" + self.settings["couch_server"].split("@")[1]
         )
+        field_sources["sourcedb_url"] = "Genomics Status (backend)"
+
+        summary_row.value["field_sources"] = field_sources
+
+        reports = {}
+        type_to_name = {
+            "_": "MultiQC",
+            "_qc_": "QC MultiQC",
+            "_pipeline_": "Pipeline MultiQC",
+        }
+        for report_type in self.get_multiqc(project, read_file=False).keys():
+            report_name = type_to_name.get(report_type, report_type)
```

Comment:
`type_to_name.get(report_type, report_type)` What does this get do?

Reply:
I've added a comment now.